### PR TITLE
Video widget

### PR DESCRIPTION
**Pull request by satbir121**

### CODE REVIEW

**Review by riodeuno — DISMISSED** [review]

File: app/client/src/widgets/VideoWidget.tsx [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+  }
+
+  static getDefaultPropertiesMap(): Record<string, string> {
+    return {};
+  }
+
+  static getTriggerPropertyMap(): TriggerPropertiesMap {
+    return {
+      // onRowSelected: true,
+      // onPageChange: true,
```
Comment (Nikhil-Nandagopal) [inline app/client/src/widgets/VideoWidget.tsx:45]: @satbir121 please remove these
Also can we have an onPlayStateChange callback

File: app/client/src/widgets/VideoWidget.tsx [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+      url: VALIDATION_TYPES.TEXT,
+    };
+  }
+
+  static getMetaPropertiesMap(): Record<string, any> {
+    return {};
+  }
+
+  static getDefaultPropertiesMap(): Record<string, string> {
+    return {};
```
Comment (Nikhil-Nandagopal) [inline app/client/src/widgets/VideoWidget.tsx:39]: @satbir121 there needs to be an internal prop for playState with values of NOT_STARTED, PAUSED, PLAYING, FINISHED

File: app/client/src/widgets/VideoWidget.tsx [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+    return {
+      ...BASE_WIDGET_VALIDATION,
+      url: VALIDATION_TYPES.TEXT,
+    };
+  }
+
+  static getMetaPropertiesMap(): Record<string, any> {
+    return {};
+  }
+
```
Comment (Nikhil-Nandagopal) [inline app/client/src/widgets/VideoWidget.tsx:37]: Need a switch property for enable auto play which will play the video on Load

File: app/client/src/widgets/VideoWidget.tsx [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+  shouldComponentUpdate(nextProps: VideoWidgetProps) {
+    return nextProps.url !== this.props.url;
+  }
+
+  getPageView() {
+    const { url } = this.props;
+    console.log({ url });
+    return (
+      <Suspense fallback={<Skeleton />}>
+        {/* <VideoComponent url="https://videowidget.s3.ap-south-1.amazonaws.com/Drone+Footage+of+a+Town.mp4"></VideoComponent> */}
```
Comment (Nikhil-Nandagopal) [inline app/client/src/widgets/VideoWidget.tsx:60]: please remove this

File: app/client/src/utils/autocomplete/EntityDefinitions.ts [hunk 5]
```diff
@@ -67,6 +67,19 @@ export const entityDefinitions = {
     isVisible: isVisible,
     searchText: "string",
   }),
+  VIDEO_WIDGET: (widget: any) => ({
+    "!doc":
+      "The Table is the hero widget of Appsmith. You can display data from an API in a table, trigger an action when a user selects a row and even work with large paginated data sets",
+    "!url": "https://docs.appsmith.com/widget-reference/table",
+    selectedRow: generateTypeDef(widget.selectedRow),
+    selectedRows: generateTypeDef(widget.selectedRows),
```
Comment (Nikhil-Nandagopal) [inline app/client/src/utils/autocomplete/EntityDefinitions.ts:75]: wrong defs for table please remove and add the devs for a video i.e playState & URL

File: app/client/src/mockResponses/PropertyPaneConfigResponse.tsx [hunk 6]
```diff
@@ -119,6 +119,22 @@ const PropertyPaneConfigResponse: PropertyPaneConfigsResponse["data"] = {
...
+    VIDEO_WIDGET: [
+      {
+        id: "17.1",
+        sectionName: "General",
+        children: [
+          {
+            id: "17.1.1",
+            propertyName: "url",
+            label: "Url",
+            controlType: "INPUT_TEXT",
```
Comment (Nikhil-Nandagopal) [inline app/client/src/mockResponses/PropertyPaneConfigResponse.tsx:131]: add properties for visibility & enable autoplay

**Review by Nikhil-Nandagopal — APPROVED** [review]

**Comment by mohanarpit** [comment]
@all-contributors Please add @satbir121 for their contributions to code and ideas.

**Pull request merged**